tests/magnetostatics/test_circuits.py

```diff
@@ -314,3 +435,77 @@ def test_cariddi(self):
         plt.close(fig)
 
         assert np.max(np.abs(ripple - self.cariddi_ripple)) < 0.04
+
+
+class TestPolyhedral2DRing:
+    @classmethod
+    def setup_class(cls):
+        radius = 7
+        n = 50
+        ring = make_circle(radius, [0, 0, 0], 0, 360, [0, 1, 0])
+        xs = Coordinates({"x": [-1, -1, 1, 1, -1], "z": [-1, 1, 1, -1, -1]})
+        xs.translate(xs.center_of_mass)
+        cls.poly_circuit = ArbitraryPlanarPolyhedralXSCircuit(
+            ring.discretize(ndiscr=n), xs, current=1e6
+        )
+        cls.arc_circuit = CircularArcCurrentSource(
+            [0, 0, 0], [1, 0, 0], [0, 0, 1], [0, 1, 0], 1, 1, 7, 360, current=1e6
+        )
```

Comment:
No need to compare with this guy, really. See `pytest tests/magnetostatics/test_circular_arc.py --plotting-on
` for an example of this being tested vs the 2-D formulation.